File: runtime/CSharp/runtime/CSharp/Antlr4.Runtime/Atn/BlockStartState.cs
/* Copyright (c) 2012-2017 The ANTLR Project. All rights reserved.
 * Use of this file is governed by the BSD 3-clause license that
 * can be found in the LICENSE.txt file in the project root.
 */
using Antlr4.Runtime.Atn;
using Antlr4.Runtime.Sharpen;

namespace Antlr4.Runtime.Atn
{
    /// <summary>
    /// The start of a regular
    /// <c>(...)</c>
    /// block.
    /// </summary>
    public abstract class BlockStartState : DecisionState
    {
        public BlockEndState endState;
    }
}

File: runtime/JavaScript/src/antlr4/state/BlockStartState.js
/* Copyright (c) 2012-2022 The ANTLR Project Contributors. All rights reserved.
 * Use is of this file is governed by the BSD 3-clause license that
 * can be found in the LICENSE.txt file in the project root.
 */
import DecisionState from './DecisionState.js';

/**
 *  The start of a regular {@code (...)} block
 */
export default class BlockStartState extends DecisionState {
    constructor() {
        super();
        this.endState = null;
        return this;
    }
}
